contract_reader.py

```diff
@@ -5,19 +5,24 @@
-address = "0x6F400810b62df8E13fded51bE75fF5393eaa841F"
-contract = web3.eth.contract(address=address, abi=abi)
 
 
-def get_current_batch_id():
+def init_contract(network='mainnet'):
+    # Specify Infura credentials
+    node_url = "https://"+network+".infura.io/v3/9408f47dedf04716a03ef994182cf150"
+    networkSwitcher = {
+        'mainnet': "0x6F400810b62df8E13fded51bE75fF5393eaa841F",
+        'rinkeby': "0xC576eA7bd102F7E476368a5E98FA455d1Ea34dE2",
+    }
+    web3 = Web3(Web3.HTTPProvider(node_url))
+    return web3.eth.contract(address=networkSwitcher.get(network, "not a valid network"), abi=abi)
```

Comment:
It would be more typesafe if network was an enum with a limited amount of cases rather than a string (which can be anything). 

Also .get is returning a default, are you sure you want this "error message" as the contract address? It sounds like an assertion would be more appropriate.

Reply:
Good catch with the .get.
 I took care of it in the other PR. 
The input correctness is already given by the choices defined by the parsing:
```
    parser.add_argument(
        '--network',
        type=str,
        choices=['mainnet', 'rinkeby'],
```
and since we don't have typesafetyness in the code in general, I think it would be overkill here.